src/node_crypto.cc

```diff
@@ -270,7 +377,10 @@ void ThrowCryptoError(Environment* env,
       .ToLocalChecked();
   CryptoErrorVector errors;
   errors.Capture();
-  auto exception = errors.ToException(env, exception_string);
+  Local<Value> exception = errors.ToException(env, exception_string);
+  Local<Object> obj;
+  if(exception->ToObject(env->isolate()->GetCurrentContext()).ToLocal(&obj))
```

Comment:
> Do you think you could propagate errors, rather than always use .ToLocalChecked()/.FromJust()

@addaleax This what you are looking for?

Reply:
@sam-github You’d typically want the inverse condition – if `.ToLocal()` returns `false`, return back to JS as fast as possible, because that condition indicates that an exception is pending. In particular, in that case we do not want to call `ThrowException()` explicitly.

(And I’m mostly concerned about the new `.FromJust()`s being added here, not so much this particular bit here.)

Reply:
@addaleax Bypassing ThrowException() if ToException() fails is a major change, and I'm struggling to see how 1) ToException could fail, 2) how this is in any way related to this PR. I'm happy to make that change in another PR, though I'd be more comfortable doing it if you could give some hints as to how I could trigger this error condition. I spent a day a while ago trying to cause "pending exceptions", and never suceeded.

Reply:
to be clear, I mean "tried in a C++ binding I wrote for playground purposes", I didn't mean with the `ToException()` call here.

Reply:
> 1) ToException could fail,

For `->Set()`, you can usually reproduce a crash by installing a setter on the target object or wrapping it in a Proxy:

```js
Object.defineProperty(Object.prototype, 'opensslErrorStack', {
  set() {
    throw new Error();
  }
});

crypto.createSign('sha1').sign(
`-----BEGIN RSA PRIVATE KEY-----
AAAAAAAAAAAA
-----END RSA PRIVATE KEY-----`)  // Currently hard-crashes
```

> 2) how this is in any way related to this PR. 

It’s not related to the subject matter of this PR, other than that I’d like for Node’s C++ code to move towards more complete error handling – not necessarily because it’s likely that conditions like the one above happen, but because a) we’ve been treating these as bugs and fixing them in the past and b) it’s really hard to see which ones *can* reasonably fail and which ones can’t, and in that case we should write code that errs on the side of caution.

It’s also not a blocking review item for this PR or anything, but I’d prefer to open a PR to address it myself afterwards if you’d like to keep this PR as it is.

Reply:
I updated. I'd like to take a shot at fixing the code you have above if you don't mind, I reproed it. Fixing it would give me a better understanding of the V8 API and how we should be dealing with errors, which is something I've struggled with, as you can see from #26929

Reply:
> I'd like to take a shot at fixing the code you have above if you don't mind, I reproed it. 

Sure! As I mentioned, we’ve fixed these kinds of things before, it’s just something that is problematic across a big part of our codebase…

Reply:
Thanks. It'll take me a few days to get to it. Hopefully, once I've done it I'll better be able to find and fix similar errors myself.